pkg/virt-handler/vm.go

```diff
@@ -2528,6 +2520,18 @@ func (c *VirtualMachineController) checkVolumesForMigration(vmi *v1.VirtualMachi
 				return true, fmt.Errorf("cannot migrate VMI with non-shared HostDisk")
 			}
 		} else {
+			isVirtiofsFilesystem := false
+			for _, filesystem := range vmi.Spec.Domain.Devices.Filesystems {
+				if filesystem.Virtiofs != nil && filesystem.Name == volume.Name {
+					isVirtiofsFilesystem = true
+					break
+				}
+			}
```

Comment:
please use the function GetFilesystemsFromVolumes instead of looping each time here